src/quest/extras/s3.py

```diff
@@ -0,0 +1,98 @@
+import os
+import json
+
+from .. import BlobStorage, Blob, WorkflowManager, WorkflowFactory, PersistentHistory, History
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+except ImportError:
+    raise ImportError("The 's3' extra is required to use this module. Run 'pip install quest-py[s3]'.")
+
+
+class S3Bucket:
```

Comment:
I followed what I did in dynamodb. Not sure if this is the best way or if we want to create some sort of root aws manager